subprojects/ide/src/test/groovy/org/gradle/plugins/ide/internal/tooling/eclipse/EclipseModelBuilderTest.groovy

```diff
@@ -66,6 +67,20 @@ class EclipseModelBuilderTest extends AbstractProjectBuilderSpec {
         eclipseModel.projectNatures.collect { it.id } == ['nature.a', 'nature.b']
     }
 
+    def "can modify natures before and after merging"() {
+        setup:
+        project.eclipse.project.natures = ['nature.a']
+        project.eclipse.project.file.beforeMerged { it.natures += 'nature.b' }
+        project.eclipse.project.file.whenMerged { it.natures += 'nature.c' }
+        def modelBuilder = createEclipseModelBuilder()
+
+        when:
+        def eclipseModel = modelBuilder.buildAll("org.gradle.tooling.model.eclipse.EclipseProject", project)
+
+        then:
+        eclipseModel.projectNatures.collect { it.id } == ['nature.b', 'nature.a', 'nature.c']
+    }
+
```

Comment:
❌ Please add Tooling API test coverage. You'll find a very similar test [here](https://github.com/gradle/gradle/blob/a0307889bed8a9ce58e1faf005a86ad86e1fb99e/subprojects/ide/src/crossVersionTest/groovy/org/gradle/plugins/ide/tooling/r30/ToolingApiEclipseModelClasspathContainerCrossVersionSpec.groovy).